### ISSUE TITLE
build failure/error using clang

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#32]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/32) by John Smith (Bitbucket: [lbalbalba](https://bitbucket.org/lbalbalba), GitHub: [lbalbalba](https://github.com/lbalbalba)).**

----------------------------------------

When I compile latest sources with llvm/clang, I run into the following build failure/error:

```
#!shell

[  1%] Building CXX object nel/src/misc/CMakeFiles/nelmisc.dir/cdb_check_sum.cpp.o
In file included from /usr/local/src/ryzomcore/code/nel/src/misc/cdb_check_sum.cpp:19:0:
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:63:23: error: âvectorâ in namespace âstdâ does not name a type
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:63:28: error: ISO C++ forbids declaration of âparameterâ with no type [-fpermissive]
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:63:34: error: expected â,â or â...â before â<â token
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h: In member function âvoid NLMISC::CCDBCheckSum::addVector(int)â:
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:22: error: âvectorâ in namespace âstdâ does not name a type
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:28: error: expected unqualified-id before â<â token
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:28: error: expected â;â before â<â token
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:28: error: expected primary-expression before â<â token
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:30: error: expected primary-expression before â>â token
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:31: error: â::const_iteratorâ has not been declared
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:48: error: expected â;â before âitâ
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:48: error: âitâ was not declared in this scope
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:53: error: âvectâ was not declared in this scope
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:65: error: expected â)â before â;â token
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:67: error: âitâ was not declared in this scope
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:73: error: âvectâ was not declared in this scope
/usr/local/src/ryzomcore/code/nel/include/nel/misc/cdb_check_sum.h:65:89: error: expected â;â before â)â token
make[2]: *** [nel/src/misc/CMakeFiles/nelmisc.dir/cdb_check_sum.cpp.o] Error 1
make[1]: *** [nel/src/misc/CMakeFiles/nelmisc.dir/all] Error 2
make: *** [all] Error 2

```

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

fixes #32 build failure/error using clang

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

I think my commit should have fixed this error, but I'm not sure because I don't have it under my Mac
